### DISCUSSION TITLE
✅ Can it parse RSA 3072 bits SSH key?

### DISCUSSION THREAD

**Discussion by cyclue** [isontheline/pro.webssh.net#1115] [Q&A]
When I use RSA 3072 bits SSH key to connect the server, it shows 'Unable to parse the private key'. I just wonder whether it support RSA 3072 bits SSH key or I just input a wrong key. By the way, the key was generated by xshell on PC.

**Comment by isontheline** [comment]
I would like to think that the format isn't supported (not the 3072 bits themself).
Could you post a truncated example of an example key generated by xshell ?

**Comment by cyclue** [comment]
Here is the key I just generated

ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAYEAvYqOl/Brw+PUytfkIta9krIBjlLPSBGrPYbe0t+DP+MQiTXt3wF5AW7TbOhSjGL1l9EV21kITuLbZAXyBTYHa4sVzKvzayPg06vWnSczP6QJ91X/vSc3mPdC9u5s1aryE79+pIQHe608qMXDbu2aMEYG4THN5ottiyDpyvVfERk6XDjreGODwMrUfQN8CK2jG6dizPNHY+aWysbtHqA/zMd0jtPBiQDAgoS6zJ6fw4n0as0LIU/XFhQVtlR1CaWpXU+XQVe6R1W6uEvGlyO8GaCVTfTp2ltb5BWGy4BNJC3pyh4jNxTzbWvHHEcjoDOxLqvojzxG18lafs0V0MORWyPiObIX2oZhhOT95qwY2iJEniCLnVI3Xp4O6OROH0Jnk0oMtSmXQBAdBnd1uo4bDf7Uv3cAXaA3EMgHVX3qiBwkCuey4uSzSDJwrWEDx/a3EnYrz6/nJ3rwXyGT6EU/zzfc0Dx0NHL1PVqAYA+S8PL7u12WP4Y0199ZC9SO8DSX

**Reply by cyclue** [reply]
The key format seems to be OpenSSH
![image](https://github.com/isontheline/pro.webssh.net/assets/104609052/05688c2a-d427-4a61-a21e-b2fa882093bc)

**Reply by isontheline** [reply]
Public Key can't be used for connection, you must use a Private Key instead.
You can generate a new Private Key from WebSSH itself.
